src/app/forms/components/date/date.component.ts

```diff
@@ -142,8 +143,16 @@ export class DateComponent extends BaseControlComponent implements OnInit, OnDes
   }
 
   processDate(year: any, month: any, day: any, hour: any, minute: any, second: any) {
-    if (this.isoDate)
+    if (this.isoDate) {
+      if (year > 999 && month > 0 && day > 0 && hour > 0 && minute > 0) {
+        this.showError = true;
+      }
+
       return `${year || ''}-${this.padded(month)}-${this.padded(day)}T${this.padded(hour)}:${this.padded(minute)}:${this.padded(second)}.000`;
+    }
+    if (year > 999 && month > 0 && day > 0) {
+      this.showError = true;
+    }
     return `${year || ''}-${this.padded(month)}-${this.padded(day)}`;
   }
 
```

Comment:
I'd say would be better to extract the show/hide logic to a getter in the component. `processDate` then has the single responsibility of formatting the date.

```ts
get showError(): boolean {
  if (this.isoDate) {
    return year > 999 && month > 0 && day > 0 && hour > 0 && minute > 0
  }
  return year > 999 && month > 0 && day > 0
}
```
and your html shouldn't have to change